src/main/java/com/icthh/xm/tmf/ms/communication/channel/telegram/TelegramChannelHandler.java

```diff
@@ -0,0 +1,123 @@
+package com.icthh.xm.tmf.ms.communication.channel.telegram;
+
+import static com.icthh.xm.tmf.ms.communication.utils.LogUtil.withLog;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.icthh.xm.commons.topic.config.MessageListenerContainerBuilder;
+import com.icthh.xm.commons.topic.domain.ConsumerHolder;
+import com.icthh.xm.commons.topic.domain.TopicConfig;
+import com.icthh.xm.tmf.ms.communication.channel.ChannelHandler;
+import com.icthh.xm.tmf.ms.communication.config.ApplicationProperties;
+import com.icthh.xm.tmf.ms.communication.domain.CommunicationSpec;
+import com.icthh.xm.tmf.ms.communication.domain.CommunicationSpec.Telegram;
+import com.icthh.xm.tmf.ms.communication.domain.MessageType;
+import com.icthh.xm.tmf.ms.communication.service.TelegramService;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.listener.AbstractMessageListenerContainer;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class TelegramChannelHandler implements ChannelHandler {
+
+    private static final String DEFAULT_TELEGRAM_CONSUMER_KEY = "default";
+
+    private final ApplicationProperties applicationProperties;
+    private final KafkaTemplate<String, String> kafkaTemplate;
+    private final KafkaProperties kafkaProperties;
+    private final TelegramService telegramService;
+    private final ObjectMapper objectMapper;
+
+    @Getter
+    private Map<String, Map<String, ConsumerHolder>> tenantTelegramConsumers = new ConcurrentHashMap<>();
+
+    @Override
+    public void onRefresh(String tenantKey, CommunicationSpec spec) {
+        if (spec == null
```

Comment:
suggest to separate botConfig variable retrieving from using:
```
List<Telegram> botConfigs = Optional.ofNullable(spec)
                                                            .map(::getChannels)
                                                            .map(::getTelegram)
if(CollectionUtils.isEmpty(botCOnfig)){
 ...
return;
}
...
```

Reply:
+